### ISSUE TITLE
Prefix GraphQL/resolve metrics with Object Type name

### ISSUE THREAD

**Issue by DrewML** [newrelic/newrelic-node-apollo-server-plugin#152]
## Is your feature request related to a problem? Please describe.
We're looking for an easy way to query for usage data of a given _nested_ field. The `resolver` metrics are what we need, but they're missing data to disambiguate for common field names.

```graphql
type Foo {
   myField: String
}

type Bar {
   myField: Int
}
```

Given the following schema, both `Foo.myField` and `Bar.myField` will log their usage as a segment/metric using `GraphQL/resolve/ApolloServer/myField`. This makes it challenging to make sense of the data, because these are 2 distinct resolvers.

Please let me know if my understanding here is wrong!

## Feature Description
For `resolve` segments/metrics, use the format `GraphQL/resolve/ApolloServer/{ObjectTypeName}.{FieldName}`. In my example in the OP, the metric/segment names would become:

- `GraphQL/resolve/ApolloServer/Foo.myField`
- `GraphQL/resolve/ApolloServer/Bar.myField`

## Describe Alternatives
Probably 3 potential implementations:

1. Breaking change to update the format of `GraphQL/resolve` segments/metrics
1. Add a configuration option to allow customization of metric/segment formats
1. Create a new segment/metric name and make it additive

## Additional context
Add any other context here.

## Priority
Please help us better understand this feature request by choosing a priority from the following options:
Really want

**Comment by michaelgoin** [comment]
@DrewML Thank you for the feature request.

This was a trade-off we were considering when creating the metrics. Removing the full path allows for seeing pattern of resolver impact when leveraged in different scenarios but obviously introduced the limitation you've hit.

We'll take a look and consider the trade-offs.

**Comment by DrewML** [comment]
Circling back on this.

@michaelgoin I think there might be some slight confusion, hopefully I can clarify.

> Removing the full path allows for seeing pattern of resolver impact when leveraged in different scenarios but obviously introduced the limitation you've hit.

This totally makes sense to me. I like that y'all have 2 classes of metrics, so you can see impact on a resolver used in multiple paths in the graph vs that resolver's usage anywhere.

But, the current `resolve` metrics don't quite accomplish that second part. For example, in GraphQL, it's very common to have a List field with the name `items`:

```
query GetProducts {
   products {
      items {
         name
      }
   }
}
```

In this example, `items` is a resolver I'd like to trace. However, if I _also_ have another field in my schema called `items`, the metrics get skewed:

```
query GetCategories {
   categories {
      items {
         name
      }
   }
}
```

In the example above, the metrics for both these `items` resolvers are collapsed into a single metric, even though they're 2 different functions that have no overlap.

In the example above, I'd expect the field to be reported as something like `Categories.items` and `Products.items`. Prepending the type name to the field name will ensure that you get metrics for _anywhere_ that field is used (because the parent type doesn't change per request like the "deep path" does).

**Comment by michaelgoin** [comment]
@DrewML Thank you for the additional details. I definitely get where you are coming from and agree some additional identification seems warranted.

**Comment by jordigh** [comment]
Fixed by #247

**Comment by bizob2828** [comment]
Hi @DrewML this has been released as part of [3.1.0](https://github.com/newrelic/newrelic-node-apollo-server-plugin/releases/tag/v3.1.0) of the plugin.  There will be new metrics called `GraphQL/typedResolve/ApolloServer/<parentType>.<resolver>` to clarify the ambiguity of resolver fields having the same name
